server/server.js

```diff
@@ -1,9 +1,30 @@
 const express = require('express');
 const bodyParser = require('body-parser');
 const path = require('path');
+const socket = require('socket.io');
 const app = express();
 const PORT = 3000;
 
+const server = app.listen(PORT, () => {
+  console.log(`Listening on port ${PORT}`);
+});
+
+const io = socket(server);
+
+// test for connection
+io.on('connection', socket => {
+  console.log('a user connected');
+  socket.on('disconnect', () => {
+    console.log('user disconnected');
+  });
+
+  // handle coding event
+  socket.on('coding', data => {
+    console.log(data);
+    socket.broadcast.emit('receive code', data);
+  });
+});
+
 // Handle parsing request body
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
```

Comment:
In my APC today Brian told me bodyParser is deprecated, said we could just use express -> `app.use(express.urlencoded({ extended: true }));
app.use(express.json());`